components/resource_metering/src/cpu/mod.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+mod collector;
+pub use collector::Collector;
+
+mod future_ext;
+pub use future_ext::FutureExt;
+
+use crate::ResourceMeteringTag;
+
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use collections::HashMap;
+
+#[cfg(target_os = "linux")]
+mod linux;
+#[cfg(target_os = "linux")]
+pub use linux::{init_recorder, register_collector, CollectorHandle, Guard};
+
+#[cfg(not(target_os = "linux"))]
+mod dummy;
+#[cfg(not(target_os = "linux"))]
+pub use dummy::{init_recorder, register_collector, CollectorHandle, Guard};
+
+#[derive(Debug, Clone)]
```

Comment:
Is it possible to **not** derive the `Clone` for it? (This will help us find places where we mistakenly cloned the content, instead of cloning the Arc reference).